File: zd3.go
package main

import (
	"fmt"
)

func isPrime(n int) bool {
	if n <= 1 {
		return false
	}
	for i := 2; i*i <= n; i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func main() {
	var numbers []int
	var num int

	// Ввод чисел
	fmt.Println("Введите числа (для завершения введите отрицательное число):")
	fmt.Scan(&num)

	for {
		if num > 0 {
			numbers = append(numbers, num)
			fmt.Scan(&num)
			if num < 0 {
				break
			}
		} else {
			fmt.Println("Нужно вводить только числа:")
			return
		}
	}

	// Проверка чисел на простоту
	fmt.Println("Результаты проверки чисел на простоту:")
	for _, number := range numbers {
		if isPrime(number) {
			fmt.Printf("%d - простое число\n", number)
		} else {
			fmt.Printf("%d - не является простым числом\n", number)
		}
	}
}

File: zd3.py
import math
import sys
def is_prime(n):
    if n <= 1:
        return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

def main():
    numbers = []
    num = int(input("Введите числа (для завершения введите отрицательное число):\n"))
    

    while True:
        try:
            if isinstance(num, (int, float)):
                if num < 0:
                    break
                numbers.append(num)
                num = int(input())
            else:
                print("Нужно вводить только числа:")
             
        except:
            sys.exit('Error')
    print("Результаты проверки чисел на простоту:")
    for number in numbers:
        if is_prime(number):
            print(f"{number} - простое число")
        else:
            print(f"{number} - не является простым числом")

if __name__ == "__main__":
    main()
